Zend/zend_compile.c

```diff
@@ -5767,9 +5772,10 @@ void zend_compile_params(zend_ast *ast, zend_ast *return_type_ast) /* {{{ */
 					"Variadic parameter cannot have a default value");
 			}
 		} else if (default_ast) {
-			/* we cannot substitute constants here or it will break ReflectionParameter::getDefaultValueConstantName() and ReflectionParameter::isDefaultValueConstant() */
+			zend_string *default_value_string = zend_new_interned_string(zend_ast_export("", default_ast, ""));
+			arg_info->default_value = ZSTR_VAL(default_value_string);
```

Comment:
You can't store the data pointer of a zend_string like that. Either you need to store a `char*` allocation, or the whole `zend_string`. See for example the distinction between `zend_internal_arg_info` and `zend_arg_info`, where the former stores the parameter name as `const char*` and the latter as `zend_string*`. This is the approach I would recommend. You would have to then switch between internal/userland representation at use-sites.

Reply:
Oh, I haven't noticed the distinction (but I couldn't have figured out the solution alone). Thank you for the explanation!

Reply:
"Done". That is, I followed how `zend_arg_info->name` is stored, but it still leaks. I'll have to learn a lot more about string handling...

Reply:
You'll also have to free the string, the code for that should be somewhere in `zend_opcode.c` (destroy_op_array or so)